modules/statistics/CallStats.js

```diff
@@ -68,164 +70,330 @@ const reportType = {
-});
 
-/* eslint-enable no-invalid-this */
+    /**
+     * Wraps some of the CallStats API method and logs their calls with
+     * arguments on the debug logging level. Also wraps some of the backend
+     * methods execution into try catch blocks to not crash the app in case
+     * there is a problem with the backend itself.
+     * @param {callstats} theBackend
+     * @private
+     */
+    static _traceAndCatchBackendCalls(theBackend) {
+        const tryCatchMethods = [
+            'associateMstWithUserID',
+            'sendFabricEvent',
+            'sendUserFeedback'
+
+            // 'reportError', - this one needs special handling - see code below
+        ];
+
+        for (const methodName of tryCatchMethods) {
+            const originalMethod = theBackend[methodName];
+
+            theBackend[methodName] = function(...theArguments) {
+                try {
+                    return originalMethod.apply(theBackend, theArguments);
+                } catch (e) {
+                    GlobalOnErrorHandler.callErrorHandler(e);
+                }
+            };
+        }
+        const debugMethods = [
+            'associateMstWithUserID',
+            'sendFabricEvent',
+            'sendUserFeedback'
 
-// some errors/events may happen before CallStats init
-// in this case we accumulate them in this array
-// and send them to callstats on init
-CallStats.reportsQueue = [];
+            // 'reportError', - this one needs special handling - see code below
+        ];
 
-/**
- * Whether the library was successfully initialized using its initialize method.
- * And whether we had successfully called addNewFabric.
- * @type {boolean}
- */
-CallStats.initialized = false;
+        for (const methodName of debugMethods) {
+            const originalMethod = theBackend[methodName];
 
-/**
- * Whether we are in progress of initializing.
- * @type {boolean}
- */
-CallStats.initializeInProgress = false;
+            theBackend[methodName] = function(...theArguments) {
+                logger.debug(methodName, theArguments);
+                originalMethod.apply(theBackend, theArguments);
+            };
+        }
+        const originalReportError = theBackend.reportError;
+
+        /* eslint-disable max-params */
+        theBackend.reportError
+        = function(pc, cs, type, error, ...otherArguments) {
+            const allArguments = [ pc, cs, type, error ].concat(otherArguments);
+
+            // Logs from the logger are submitted on the applicationLog event
+            // "type". Logging the arguments on the logger will create endless
+            // loop, because it will put all the logs to the logger queue again.
+            if (type === wrtcFuncNames.applicationLog) {
+                console && console.debug('reportError', allArguments);
+            } else {
+                logger.debug('reportError', allArguments);
+            }
+            try {
+                originalReportError.apply(theBackend, allArguments);
```

Comment:
Instead of using an array you can use `.call` and `...args`. Also, you don't seem to use the error parameter, so you can leave it as part of `...args`.